### PR TITLE
Mat 7689 assign test case no when drafting

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-7689](https://jira.cms.gov/browse/MAT-7689)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/services/VersionService.java [hunk 1]
```diff
@@ -252,10 +279,21 @@ private List<TestCase> cloneTestCases(List<TestCase> testCases, List<Group> draf
...
+                        .groupPopulations(updatedTestCaseGroupPopulations)
+                        .build();
+                if (isFlagEnabled) {
+                  if (caseNumberExists) {
+                    testCase.setCaseNumber(testCase.getCaseNumber());
+                  } else {
+                    testCase.setCaseNumber(sequenceService.generateSequence(measureId));
+                  }
+                } else {
+                  testCase.setCaseNumber(null);
```
Comment (jkotanchik-SB) [inline src/main/java/cms/gov/madie/measure/services/VersionService.java]: Does setting `caseNumber` to `null` differ from not setting it at all? If not, then I think this `else` can be removed.

Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/VersionService.java]: @jkotanchik-SB so the caseNumber will be set before the if flagEnabled, I just wanted to make sure if the flag is not enabled, we don't set any value, then again, there probably wouldn't be any value before the flag is enabled, so maybe this is over kill, I'll remove it (I added this part right before the check in)

File: src/main/java/cms/gov/madie/measure/services/VersionService.java [hunk 2]
```diff
@@ -356,4 +394,17 @@ private void savePackageData(Measure savedMeasure, byte[] packageData, String us
         username,
         savedExport.getId());
   }
+
+  private boolean checkCaseNumberExists(List<TestCase> testCases, String measureId) {
```
Comment (jkotanchik-SB) [inline src/main/java/cms/gov/madie/measure/services/VersionService.java]: `measureId` isn't used in method body. Can it be removed?

File: src/main/java/cms/gov/madie/measure/services/VersionService.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/VersionService.java]: @jkotanchik-SB removed

File: src/main/java/cms/gov/madie/measure/services/VersionService.java (hunk 2 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/VersionService.java]: @jkotanchik-SB removed

**Review by jkotanchik-SB — APPROVED** [review]
